plugins/context_datalayer_reaction.inc

```diff
@@ -170,6 +170,6 @@ class context_datalayer_reaction extends context_reaction {
 /**
  * AJAX callback for Context reaction options form.
  */
-function context_datalayer_options_form_ajax($form, $form_state) {
-  return $form['reactions']['plugins']['context_datalayer']['data'];
+function context_datalayer_reaction_options_form_ajax($form, $form_state) {
+  return $form['reactions']['plugins']['context_datalayer_reaction']['context_datalayer_reaction']['data'];
```

Comment:
Notice the massive difference between the two arrays?  Context module seems to do some nonsense behind the scenes which messes this up completely.  Being explicit means we have to point exactly to the one we want but at least there's no confusion now.  The worst case I think I had was a reaction wrapper around a condition form ... It's puzzling how that could ever have happened but it did!